Gibe.AbTest/AbTestingService.cs

```diff
@@ -17,7 +17,8 @@ public IEnumerable<Experiment> GetExperiments()
 			var experiments = _abTestRepository.GetExperiments()
 				.Where(e => e.Enabled)
 				.Select(e => new Experiment(e, GetAvailableVariations(e.Id).ToArray()))
-				.Where(e => e.Variations.Any());
+				.Where(e => e.Variations.Any())
+				.ToArray();
```

Comment:
I originally used `ToList()` but then changed to `ToArray()` as I saw it being used on the line above and wanted to be consistent.  Ironically, I wondered if the reason that `ToArray()` had been used is because it might be more performant in the eyes of the original coder (I dunno...it _feels_ like it should be as it's a simpler data type...though it's more primitive - Guess I should Google it).

I don't have any strong feelings on it either way, so could change my work + line above to `ToList()`?

Reply:
Ha - my response was to your original comment that was in my notification email!

I've just doing some reading on a [_really long_ SO thread](https://stackoverflow.com/questions/1105990/is-it-better-to-call-tolist-or-toarray-in-linq-queries) - I'm guessing this is the same thread you read which has many conflicting views?

One of the other reasons to choose `ToArray()` is because we're never going to be changing the list once it's out of the database - it's purely for reading which I think was also at the back of my mind when I saw the other `ToArray()`.  So maybe that's why the original developer put `ToArray()` in the `Select()`?

Anyways - I think I'll stick with `ToArray()` for now just because I don't think it'll make much performance difference and it "feels" more right because it's readonly (I know..I can put something like `ToList().AsReadOnly()`, but right now I feel like it'd be splitting hairs 😂 )